pkg/datapath/linux/route/reconciler/manager.go

```diff
@@ -0,0 +1,273 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package reconciler
+
+import (
+	"errors"
+	"fmt"
+	"slices"
+
+	"github.com/cilium/statedb"
+	"github.com/cilium/statedb/reconciler"
+
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+type DesiredRouteManager struct {
+	db  *statedb.DB
+	tbl statedb.RWTable[*DesiredRoute]
+
+	mu     lock.Mutex
+	owners map[string]*RouteOwner
+}
+
+func newDesiredRouteManager(db *statedb.DB, tbl statedb.RWTable[*DesiredRoute]) *DesiredRouteManager {
+	return &DesiredRouteManager{
+		db:     db,
+		tbl:    tbl,
+		owners: make(map[string]*RouteOwner),
+	}
+}
+
+var (
+	ErrOwnerExists       = errors.New("owner already exists")
+	ErrOwnerDoesNotExist = errors.New("owner does not exist")
+)
+
+func (m *DesiredRouteManager) RegisterOwner(name string, adminDistance AdminDistance) (*RouteOwner, error) {
```

Comment:
I'd suggest binding AD to the routes instead of owners. If we add BGP route import support in the future for example, each BGP instance would be a owner, but the same BGP instance would install iBGP and eBGP routes which should have different AD values.